src/pygeoprocessing/routing/routing.pyx

```diff
@@ -2960,3 +3006,1287 @@ def _is_raster_path_band_formatted(raster_path_band):
         return False
     else:
         return True
+
+
+def extract_strahler_streams_d8(
+        flow_dir_d8_raster_path_band, flow_accum_raster_path_band,
+        dem_raster_path_band,
+        target_stream_vector_path,
+        long min_flow_accum_threshold=100,
+        int river_order=5,
+        float min_p_val=0.05,
+        osr_axis_mapping_strategy=DEFAULT_OSR_AXIS_MAPPING_STRATEGY):
+    """Extract Strahler order stream geometry from flow accumulation.
+
+    Creates a Strahler ordered stream vector containing line segments
+    representing each separate stream fragment. The final vector contains
+    at least the fields:
+        * "order" (int): an integer representing the stream order
+        * "river_id" (int): unique ID used by all stream segements that
+            connect to the same outlet.

```

Comment:
Weeeelll... I did say "at least the following fields" because in a way they're useful for something else, but I guess that should be documented if it really is useful huh?

Reply:
Okay i documented all the features in 0903796a7b389b1123f4176f62774c1ca9f82f30 except "d8_flow_dir" which was just an intermediate field anyway that I delete now before the function is over.